fix(dht, trackerless-network): Strict boolean expressions

teogeb opened the pull request:
Use strict boolean expressions, i.e. `if (foo !== undefined)` instead of `if (foo)`. 

Some expressions were fixed and some issues were annotated with TODO comments.

## Background

There has been several problems related to boolean expression which handle nullable values in checks/assertions:
- currently we have some boolean expressions which are unintentionally always `true`
  - see "`better check (currently this condition is always true)`" comments added in this PR
- there was recently a problem when we used nullable `HandshakeError` enum https://github.com/streamr-dev/network/pull/1904
  - in many enums on one of the values is `0`, and therefore lenient `null` checks can incorrectly ignore the value
 - the `MigrateData` test is broken https://github.com/streamr-dev/network/pull/2088

## Future improvements

- These warnings were found with https://typescript-eslint.io/rules/strict-boolean-expressions/ `eslint` rule. We could enable the rule if we add TypeScript support to our `eslint` setup (see https://github.com/streamr-dev/network/pull/1171). 
- Could also  run the `eslint` rule with `allowString=false` and `allowNumber=false` to get some more potential places to fix?
- Could also change some `if` statements which use objects to have explicit undefined check (i.e. `if (obj)` to `if (obj !== undefined`)) 
  - use `if \([!a-zA-Z0-9.]+\)` regexp to find those

teogeb committed:
strict boolean expressions

juslesan reviewed (approved):
Opening PR's for TODO comments ASAP

juslesan committed:
fix(dht): Get contact typing fix (#2091)

## Summary

Fix typing of `Contact#getContact`, the function could return
`undefined` but the `!` operator made build pass unexpectedly.
